### ISSUE TITLE
Cannot run action from a different directory

### ISSUE THREAD

**Issue by jwnx** [pulumi/actions#35]
_* Edited to match documentation (https://www.pulumi.com/docs/guides/continuous-delivery/github-actions/#using-a-different-root-directory)_

The recommended way of running an action from a different directory is by setting `PULUMI_ROOT` to the relative directory for a Pulumi project. Since my setup includes a root-level `package.json` and `tsconfig.json`, I need to copy both files to the subdirectory (https://github.com/pulumi/pulumi/issues/2619). But doing so doesn't work:

```
Logging in using access token from PULUMI_ACCESS_TOKEN
Logged in to pulumi.com as juliana
Activated service account credentials for: [***.iam.gserviceaccount.com]
Adding credentials for all GCR repositories.
WARNING: A long list of credential helpers may cause delays running 'docker build'. We recommend passing the registry name to configure only the registry you are using.
Docker configuration file updated.
yarn install v1.22.5
warning package.json: No license field
warning infrastructure@0.0.1: No license field
[1/4] Resolving packages...
[2/4] Fetching packages...
[3/4] Linking dependencies...
[4/4] Building fresh packages...
Done in 22.41s.
#### :tropical_drink: `pulumi --non-interactive preview --suppress-outputs`
Previewing update (...)

View Live: https://app.pulumi.com/**/previews/73e3a996-7730-46b5-bc73-3509ee94b100

    pulumi:pulumi:Stack infrastructure  error: It looks like the Pulumi SDK has not been installed. Have you run npm install or yarn install?
    pulumi:pulumi:Stack infrastructure  1 message
 
Diagnostics:
  pulumi:pulumi:Stack (infrastructure):
    error: It looks like the Pulumi SDK has not been installed. Have you run npm install or yarn install?
```

The workflow file:

```
name: Pulumi
on:
  pull_request:
    types: [opened, reopened, synchronize]
    branches:
      - master

jobs:
  preview:
    name: Preview
    runs-on: self-hosted
    steps:
      - uses: actions/checkout@v2
        with:
          fetch-depth: 0

      - run: mv package.json cluster/package.json && mv yarn.lock cluster/yarn.lock && mv tsconfig.json cluster/tsconfig.json

      - uses: docker://pulumi/actions
        with:
          args: preview --suppress-outputs
        env:
          PULUMI_ACCESS_TOKEN: ${{ secrets.PULUMI_ACCESS_TOKEN }}
          GOOGLE_CREDENTIALS: ${{ secrets.GCLOUD_SERVICE_ACCOUNT }}
          COMMENT_ON_PR: 1
          GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
          PULUMI_CI: pr
          PULUMI_ROOT: cluster
```

Locally, I usually run  `--cwd` to run Pulumi from another directory and works without issues. I also tested running with `--cwd` on Pulumi's docker image and it works as expected. i.e.:

```
docker run --rm --name pulumiactions --label foo --workdir /github/workspace -e GITHUB_ACTIONS=true -e GITHUB_WORKFLOW="test" -e PULUMI_CI="pr" -e PULUMI_ACCESS_TOKEN=$PULUMI_ACCESS_TOKEN -e GOOGLE_CREDENTIALS=$GOOGLE_CREDENTIALS -e CI=true -v "$(pwd)":"/github/workspace" pulumi/actions preview --cwd ./cluster --stack stackname --suppress-outputs
```

But running on Github Actions, it gives me a `failed to locate the entry point for your program: /github/workspace` error.

```
#### :tropical_drink: `pulumi --non-interactive preview --cwd ./cluster --stack stackname --suppress-outputs`
Previewing update (stackname)

View Live: ...

    pulumi:pulumi:Stack infrastructure running 
    pulumi:pulumi:Stack infrastructure running We failed to locate the entry point for your program: /github/workspace
    pulumi:pulumi:Stack infrastructure running Here's what we think went wrong:
    pulumi:pulumi:Stack infrastructure running   * Your program looks like a TypeScript program. Have you run 'tsc'?
    pulumi:pulumi:Stack infrastructure  3 messages
```

The workflow file: 

```
name: Pulumi
on:
  pull_request:
    types: [opened, reopened, synchronize]
    branches:
      - master

jobs:
  preview:
    name: Preview
    runs-on: self-hosted
    steps:
      - uses: actions/checkout@v2
        with:
          fetch-depth: 1

      - uses: docker://pulumi/actions
        with:
          args: preview --cwd "./cluster" --stack stackname --suppress-outputs
        env:
          PULUMI_ACCESS_TOKEN: ${{ secrets.PULUMI_ACCESS_TOKEN }}
          GOOGLE_CREDENTIALS: ${{ secrets.GCLOUD_SERVICE_ACCOUNT }}
          COMMENT_ON_PR: 1
          GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
          PULUMI_CI: pr
```
 
So far I haven't been able to identify why any of these options aren't working and how to make Pulumi work from a subdirectoy.

**Comment by stack72** [comment]
Closed as being tracked in #40

**Comment by stack72** [comment]
I am closing this out as we have just released v2 of our Action which is based on TypeScript and addresses this issue
